frontend/src/app/spindle/spindle.component.ts

```diff
@@ -39,41 +58,61 @@ export class SpindleComponent implements OnInit, OnDestroy {
                     this.errorHandler.handleSpindleError(response.error);
                     return;
                 }
-                if (response.tex) {
-                    this.texOutput = response.tex;
+                if (response.latex) {
+                    this.textOutput = {
+                        extension: 'tex',
+                        text: response.latex
+                    }
                 }
                 if (response.redirect) {
                     // Opens a new tab.
                     window.open(response.redirect, "_blank");
                 }
                 if (response.pdf) {
-                    this.downloadPDF(response.pdf);
+                    const base64 = response.pdf;
+                    const linkSource = `data:application/pdf;base64,${base64}`;
+                    const fileName = "spindleParseResult.pdf";
+                    this.downloadFile(fileName, linkSource);
```

Comment:
why isn't this using `downloadAsFile()`?

Reply:
The reason was that the base64 string returned from the server does not need to be Blobified (while .tex does). I agree it would be nicer to use `downloadAsFile()` for PDFs as well, so I have updated it.